standard_renderer.go

```diff
@@ -179,95 +185,69 @@ func (r *standardRenderer) flush() {
-		}
-	}
-
-	// Merge the set of lines we're skipping as a rendering optimization with
-	// the set of lines we've explicitly asked the renderer to ignore.
-	for k, v := range r.ignoreLines {
-		skipLines[k] = v
-	}
-
 	if flushQueuedMessages {
-		// Dump the lines we've queued up for printing
+		// Dump the lines we've queued up for printing.
 		for _, line := range r.queuedMessageLines {
+
+			// Removing previousy rendered content at the end of line.
+			line = line + ansi.EraseLineRight
+
 			_, _ = buf.WriteString(line)
 			_, _ = buf.WriteString("\r\n")
 		}
-		// clear the queued message lines
+		// Clear the queued message lines.
 		r.queuedMessageLines = []string{}
 	}
 
-	// Paint new lines
+	// Paint new lines.
 	for i := 0; i < len(newLines); i++ {
-		if _, skip := skipLines[i]; skip {
+		canSkip := !flushQueuedMessages && // Queuing messages triggers repaint -> we don't have access to previous frame content.
+			len(oldLines) > i && oldLines[i] == newLines[i] // Previously rendered line is the same.
+
+		if _, ignore := r.ignoreLines[i]; ignore || canSkip {
 			// Unless this is the last line, move the cursor down.
 			if i < len(newLines)-1 {
 				buf.WriteString(ansi.CursorDown1)
 			}
-		} else {
-			if i == 0 && r.lastRender == "" {
-				// On first render, reset the cursor to the start of the line
-				// before writing anything.
-				buf.WriteByte('\r')
-			}
+			continue
+		}
 
-			line := newLines[i]
-
-			// Truncate lines wider than the width of the window to avoid
-			// wrapping, which will mess up rendering. If we don't have the
-			// width of the window this will be ignored.
-			//
-			// Note that on Windows we only get the width of the window on
-			// program initialization, so after a resize this won't perform
-			// correctly (signal SIGWINCH is not supported on Windows).
-			if r.width > 0 {
-				line = ansi.Truncate(line, r.width, "")
-			}
+		if i == 0 && r.lastRender == "" {
+			// On first render, reset the cursor to the start of the line
+			// before writing anything.
+			buf.WriteByte('\r')
+		}
 
-			_, _ = buf.WriteString(line)
+		line := newLines[i]
 
-			if i < len(newLines)-1 {
-				_, _ = buf.WriteString("\r\n")
-			}
+		// Removing previousy rendered content at the end of line.
+		line = line + ansi.EraseLineRight
+
+		// Truncate lines wider than the width of the window to avoid
+		// wrapping, which will mess up rendering. If we don't have the
+		// width of the window this will be ignored.
+		//
+		// Note that on Windows we only get the width of the window on
+		// program initialization, so after a resize this won't perform
+		// correctly (signal SIGWINCH is not supported on Windows).
+		if r.width > 0 {
+			line = ansi.Truncate(line, r.width, "")
+		}
+
+		_, _ = buf.WriteString(line)
+
+		if i < len(newLines)-1 {
+			_, _ = buf.WriteString("\r\n")
 		}
 	}
+
+	// Clearing left over content from last render.
+	if r.linesRendered > numLinesThisFlush {
+		buf.WriteString(ansi.EraseDisplayBelow)
```

Comment:
```suggestion
		buf.WriteString(ansi.EraseScreenBelow)
```

Reply:
bumped `ansi` version in main?

Reply:
Yes, I just released a v1.1.2 to include the `x/ansi` changes